app/helpers/proposal_details_helper.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+module ProposalDetailsHelper
+  def proposal_details_group_id(group)
+    proposal_details_group_name(group).downcase.gsub(/[^0-9a-z]/i, "")
+  end
+
+  def proposal_details_group_title(group)
+    proposal_details_group_name(group).downcase.underscore.humanize
+  end
+
+  def proposal_details_group_name(group)
+    group || t("proposal_details.other")
+  end
+
+  def proposal_detail_item(proposal_detail)
```

Comment:
I've moved all these methods out of `ProposalDetailsPresenter` as that was essentially defining them on `planning_application` which doesn't really make sense. This change means that instead of 

```
planning_application.proposal_details.each { |proposal_detail| planning_application.my_method(proposal_detail) ```
we can write

```
planning_application.proposal_details.each { |proposal_detail| my_method(proposal_detail) }
```

Reply:
An alternative is to have an _actual_ `ProposalDetailsPresenter` that wraps a `proposal_detail` and define them there. Would that be nicer? Please tell me what you think!